src/bin/cargo-flamegraph.rs

```diff
@@ -100,51 +100,104 @@ enum Opts {
     Flamegraph(Opt),
 }
 
-fn build(opt: &Opt) -> anyhow::Result<Vec<Artifact>> {
-    use std::process::{Command, Output, Stdio};
-    let mut cmd = Command::new("cargo");
+enum RequestedTarget {
```

Comment:
I think there's way too much stuff going on in this commit, and I'm not convinced all this stuff pulls its weight. For this PR, can we cut out this commit and/or minimize it to only build the unit test with cargo test?

Reply:
I agree that the commit is too big. I created a new [branch](https://github.com/ede1998/flamegraph/commits/build-with-cargo-test) where I kept the changes to a minimum. Commit 63d9d3be0e0f479e399d9362c1f4cf097b2b21a5 introduces building unit tests with `cargo test`. However this unfortunately means that we sometimes execute `cargo metadata` twice: once when finding a unit test target and again when determining whether to call `cargo test --lib` or `cargo test --bin <bin>`. As the first call occurs in `find_unique_target`, the most elegant way to avoid that seems to be to add the `kind` to the return value `BinaryTarget`, which is what I was trying to do in the mega commit. Though I can probably cut quite a bit of stuff out now that I have a look at it again.

Reply:
I'd be fine with calling `cargo metadata` twice for now and then separately following up on improvements to that.

Reply:
Ok, I tried a third implementation in [this branch](https://github.com/ede1998/flamegraph/tree/parse-build-output-2). Commit dad539e9fe372466822dc5d59488067601b1a441 is another minimal implementation that builds unit test with `cargo test`. This one avoids running `cargo metadata` twice though but introduces an `kind` parameter to the build method which is only used when building a unit test target. I tried passing the `BinaryTarget in instead but that seems to massively bloat the commit again. So there is 3 versions now:
- Hopelessly bloated commit 346ed3281c4fb48d4c1ce2fe1711679b4a8371f8 in branch [ede1998:profile_unit_tests](https://github.com/ede1998/flamegraph/tree/profile_unit_tests)
- Running `cargo metadata` twice commit 63d9d3be0e0f479e399d9362c1f4cf097b2b21a5 in branch [ede1998:build-with-cargo-test](https://github.com/ede1998/flamegraph/tree/build-with-cargo-test)
- Extra `kind` parameter in build commit dad539e9fe372466822dc5d59488067601b1a441 in branch [ede1998:parse-build-output-2](https://github.com/ede1998/flamegraph/tree/parse-build-output-2) (whoops, that's not the branch name i was going for but it doesn't matter)

Which version do you like best?

Reply:
The extra `kind` parameter version looks good to me! Thanks for going through this comparison!

Reply:
Ok, I moved the commits into this PR and rebased them onto main.
Additionally, I made a small change to commit 707121d790b5ffad348f3c92c71e15a70536bd09: `trailing_arguments` now has a doc string and I updated the USAGE section in README again, which fixed two typos: `performace -> performance` and `default 0.1 -> 0.01`.